src/module.ts

```diff
@@ -80,7 +80,8 @@ export default defineNuxtModule<ModuleOptions>({
       lifetime: 60 * 60 * 8,
       domain: '',
       path: '/',
-      sameSite: 'lax'
+      sameSite: 'lax',
+      clientOnly: false,
```

Comment:
Instead of adding a `clientOnly` option inside `cookies`, I think we should add a `ssr` option at the root level, set with `nuxt.options.ssr` value (we just want to expose it in public config to access it with `useRuntimeConfig` composable).

In this file, all server related assets (server plugin, server session route, nitro hooks, servers services) should only be register if `ssr = true`

Reply:
By *ssr option*, do you mean [the nuxt ssr option](https://nuxt.com/docs/api/configuration/nuxt-config#ssr) ?

Reply:
Indeed, this is the `nuxt.options.ssr` key. You can just use this key to conditionally register server related assets in the `module.ts`  file. Then instead of using the `useRuntimeConfig().public.supabase.cookies.clientOnly` in the `supabase.client.ts` file, you can just use the `useRuntimeConfig().ssr` value. Wdyt ?

Reply:
Hello @larbish,
Thanks for your support. You did a lot helping me to produce a viable PR. Tanks again ... but I failed :cry: 

I committed my last attempts in [my fork](https://github.com/arundo-tech/supabase/tree/138_static_hosting) in case it could help.

Once again, I feel the recurring JS/TS fatigue :disappointed:
Fixing an apparently *simple bug* (having supabase auth working on static nuxt sites whereas it already works fine in other cases) raises new troubles/questions :
* What is the preferred tooling to dev/test/debug the module in my project ? I tried pnpm workspaces, yarn link,  jump into the monorepo bandwagon to discover how crazy it is to master and fix all the package references and ressources path troubles. Too complicated for me.
* From my naive point of view (I'm not a JS/TS expert), I thought setting a cookie was kind of trivial. I'm still confused how `useCookie` and `setCookie` are so different. Having to dive into `H3Event` and take care about what could be run on client or server is also too complicated for me and my static site.
* Everything seems so *fragile* (**I** feel fragile and dumb coding like that) that I am not able to ask the real question : Why do we need a server API in this module ? How setting a cookie server side is better than client-side ? I don't see any security pros.

I hope you will not get me wrong : I **love** nuxt and its ecosystem (particularly the docus/content). Our frontend is mostly written in MDC syntax with a lot of custom component ... and it's great ... and we will continue to do so !!

We will probably copy/paste the composables/plugins of your module to our project (with credits). It is very sad to end up with this zero-level of contribution but it's too painfull.

Sorry for this free rant, I'm much more guilty than you are or any other great guys behind nuxt :wink: 
Hope this testimony makes some sense (discard it if not).
I also hope my next contribution will be more helpful.

Reply:
Hello, I would like to help and contribute to finish this PR, @PhE what is missing to finish?
I think that a module as good as this one should really be able to work in static

Reply:
Hello, 

First I'm sorry I did not answer your message @PhE. Lot's of work currently and I completely forgot about it. Just want to say that I am already glad you try and I completely understand your point. So thank you for the try ! 

Concerning the PR itself, I think you were close to make it work. To answer your questions in the last message:
1. I'm using the playground to debug and `yarn link` to use the built module in my personal project. 
2. You can use `useCookie` on frontside and `setCookie` on server side. 
3. We're using a server API to set cookies since it is more secure from server but you can do it from frontside it will work the same way. That's the reason why I want to add a condition based on the `nuxt.options.ssr`. 

I hope it helps @XStarlink. You can read the messages of this PR to understand what's missing. Don't hesitate if you have any question.

Reply:
Thanks a lot @larbish for your quick response! I started to dive into @PhE's PR and the module code to figure out how to fix the problem. I ran the playground locally and did some testing.

I understood what you say in the previous comments about 'nuxt.options.ssr' that we have to check in the module.ts to know if we use addServerHandler() or not.

I raised some questions:

- Wouldn't relying on the ssr key penalize devs who don't use ssr but still use the server part with nuxt build? (For example I don't use ssr and I still use the server part on several WebApp. So there is a case with ssr: false, where we need to use the utilities of this module on the server side like getSupabaseUser, etc... so we could continue to use the cookie on the server side)

- That being said, do we continue to rely on the ssr key or do we use a new key as proposed by @PhE?

- Do we make two cases, one where the cookies are set in Client and the other in Server or do we make only one case to simplify the use for the users?

I really want to contribute to solve this problem, but I specify that I have never developed a nuxt module, I may ask many other questions 😄

Reply:
Things cooled down a little bit and I manage to make it works without smashing my keyboard :angel: 
I had a great call with @XStarlink on this issue : I will push my [working branch](https://github.com/arundo-tech/supabase/tree/serverless) and we will work to test and clean it up.
It's client-side only for the moment. We may follow @larbish hints for the server side later.

Thanks again @larbish for nursuring the newcomer I am :clap: